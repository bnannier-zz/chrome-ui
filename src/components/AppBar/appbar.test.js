import React from "react";
import { configure, shallow, mount } from "enzyme";
import sinon from "sinon";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import { AppBar, FixedAdjust } from "../";

describe("AppBar Tests", () => {
  it("Default", function() {
    const wrapper = shallow(<AppBar>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it("Fixed", function() {
    const wrapper = shallow(<AppBar fixed>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("Dense", function() {
    const wrapper = shallow(<AppBar dense>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("Prominent", function() {
    const wrapper = shallow(<AppBar prominent>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("Short", function() {
    const wrapper = shallow(<AppBar short>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("Short Collapsed", function() {
    const wrapper = shallow(<AppBar shortCollapsed>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("Has Action Items", function() {
    const wrapper = shallow(<AppBar hasActionItems>test</AppBar>);
    expect(wrapper.render().find("header").length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });
});
