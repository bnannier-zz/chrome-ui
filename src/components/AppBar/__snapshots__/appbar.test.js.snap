// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Tests Default 1`] = `
<Base>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": undefined,
          "mdc-top-app-bar--fixed": undefined,
          "mdc-top-app-bar--prominent": undefined,
          "mdc-top-app-bar--short": undefined,
          "mdc-top-app-bar--short-collapsed": undefined,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;

exports[`AppBar Tests Dense 1`] = `
<Base
  dense={true}
>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": true,
          "mdc-top-app-bar--fixed": undefined,
          "mdc-top-app-bar--prominent": undefined,
          "mdc-top-app-bar--short": undefined,
          "mdc-top-app-bar--short-collapsed": undefined,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;

exports[`AppBar Tests Fixed 1`] = `
<Base
  fixed={true}
>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": undefined,
          "mdc-top-app-bar--fixed": true,
          "mdc-top-app-bar--prominent": undefined,
          "mdc-top-app-bar--short": undefined,
          "mdc-top-app-bar--short-collapsed": undefined,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;

exports[`AppBar Tests Has Action Items 1`] = `
<Base
  hasActionItems={true}
>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": undefined,
          "mdc-top-app-bar--fixed": undefined,
          "mdc-top-app-bar--prominent": undefined,
          "mdc-top-app-bar--short": undefined,
          "mdc-top-app-bar--short-collapsed": undefined,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;

exports[`AppBar Tests Prominent 1`] = `
<Base
  prominent={true}
>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": undefined,
          "mdc-top-app-bar--fixed": undefined,
          "mdc-top-app-bar--prominent": true,
          "mdc-top-app-bar--short": undefined,
          "mdc-top-app-bar--short-collapsed": undefined,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;

exports[`AppBar Tests Short 1`] = `
<Base
  short={true}
>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": undefined,
          "mdc-top-app-bar--fixed": undefined,
          "mdc-top-app-bar--prominent": undefined,
          "mdc-top-app-bar--short": true,
          "mdc-top-app-bar--short-collapsed": undefined,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;

exports[`AppBar Tests Short Collapsed 1`] = `
<Base
  shortCollapsed={true}
>
  <header
    className={
      Array [
        "mdc-top-app-bar",
        Object {
          "mdc-top-app-bar--dense": undefined,
          "mdc-top-app-bar--fixed": undefined,
          "mdc-top-app-bar--prominent": undefined,
          "mdc-top-app-bar--short": true,
          "mdc-top-app-bar--short-collapsed": true,
          "mdc-top-app-bar--short-has-action-item": undefined,
        },
      ]
    }
  >
    <AppBarRow>
      test
    </AppBarRow>
  </header>
</Base>
`;
